fps/src/main/scala/chapter03/exercise3_2.scala

```diff
@@ -0,0 +1,38 @@
+package chapter03
+
+import chapter03.exercise3_1.List.sum
+import chapter03.exercise3_2.List
+
+object exercise3_2 extends App {
+  sealed trait List[+A]
+  case object Nil extends List[Nothing]
+  case class Cons[+A](head: A, tail: List[A]) extends List[A]
+
+  object List {
+    def sum(ints: List[Int]): Int = ints match {
+      case Nil => 0
+      case Cons(x, xs) => x + sum(xs)
+    }
+
+    def product(ds: List[Double]): Double = ds match {
+      case Nil => 1.0
+      case Cons(0.0, _) => 0.0
+      case Cons(x, xs) => x * product(xs)
+    }
+    def getTail[A](ds: List[A]): List[A] = {
+      ds match {
+        case Nil=> Nil
+        case Cons(x,y) => y
+      }
+    }
+
+    def apply[A](as: A*): List[A] =
+      if (as.isEmpty) Nil
+      else Cons(as.head, apply(as.tail: _*))
```

Comment:
sanity check question - is apply tail recursive ???

Reply:
At the moment not. I cannot get it recursive bc . Could we look at this together?

Reply:
will address this in the one-on-ones